controller/src/large_access.rs

```diff
@@ -39,15 +39,24 @@ where
 
     /// Split a single large access into many, using `Self::build_one()`.
     fn build_many(page: P, offset: u8, len: u8) -> Result<Vec<Self>, Error> {
-        let end = offset + len;
-        (offset..end)
+        // A valid read scenario could lead the "end" to be 256 (e.g., offset = 192 and len = 64).
+        // Simply adding these as u8 would result in an overflow.
+        let stop = offset as u16 + len as u16;
+
+        // The module memory map is only 256 bytes long, reject reads that attempt to go beyond
+        // that.
+        if stop > 256 {
+            return Err(Error::ByteOutOfRange(stop as usize));
+        }
+
+        (offset as u16..stop)
             .step_by(usize::from(Self::SIZE))
             .map(|new_offset| {
                 // The length is up to SIZE, or the remainder of the entire
                 // operation, whichever is smaller.
-                let remainder = end - new_offset;
+                let remainder = (stop - new_offset) as u8;
```

Comment:
It's probably too conservative, but could we make this `u8::try_from(stop - new_offset).unwrap()`? The `as` cast makes me a bit nervous :)